chaosazure/machine/actions.py

```diff
@@ -117,6 +118,58 @@ def start_machine(filter: str = None,
     __start_stopped_machines(client, stopped_machines)
 
 
+def stress_cpu(filter: str = None,
+               duration: int = 120,
+               configuration: Configuration = None,
+               secrets: Secrets = None):
+    """
+    Stress CPU up to 100% at random machine.
+
+    Parameters
+    ----------
+    filter : str
+        Filter the virtual machines. If the filter is omitted all machines in
+        the subscription will be selected as potential chaos candidates.
+        Filtering example:
+        'where resourceGroup=="myresourcegroup" and name="myresourcename"'
+    """
+    logger.info(
+        "Start stress_cpu: configuration='{}', filter='{}'".format(
+            configuration, filter))
+
+    choice = __fetch_machine_at_random(filter, configuration, secrets)
+
+    logger.info("Stressing CPUs machine: {}".format(choice['name']))
+    client = __init_client(secrets, configuration)
+
+    os_type = __get_os_type(choice)
+    if os_type == OS_WINDOWS:
+        command_id = 'RunPowerShellScript'
+        script_name = "cpu_stress_test.ps1"
+    elif os_type == OS_LINUX:
+        command_id = 'RunShellScript'
+        script_name = "cpu_stress_test.sh"
+    else:
+        raise Exception("Cannot run CPU stress test on OS: %s" % os_type)
```

Comment:
Might be better to raise "ActivityFailed" so that the chaostoolkit doesn't fail the whole experiment

from chaoslib.exceptions import ActivityFailed

Reply:
great! I was wondering if we have some application specific exception :) will update it